xarray/backends/locks.py

```diff
@@ -49,12 +49,13 @@ class SerializableLock(Lock):
     lock: threading.Lock
 
     def __init__(self, token: Hashable | None = None):
-        self.token = token or str(uuid.uuid4())
-        if self.token in SerializableLock._locks:
-            self.lock = SerializableLock._locks[self.token]
-        else:
-            self.lock = threading.Lock()
-            SerializableLock._locks[self.token] = self.lock
+        with threading.Lock():
+            self.token = token or str(uuid.uuid4())
+            if self.token in SerializableLock._locks:
+                self.lock = SerializableLock._locks[self.token]
+            else:
+                self.lock = threading.Lock()
+                SerializableLock._locks[self.token] = self.lock
```

Comment:
If I understand correctly, the issue is with accessing the `SerializableLock._locks` registry: if two threads concurrently access that with the same token, we can get race conditions (and `uuid.uuid4` should guarantee unique tokens).

To fix that, we'd need a lock shared with all threads for the `_locks` registry (just creating a new one every time won't help) – but that's what the note in the docstring is referring to:
> The creation of locks is itself not threadsafe.

which means that we could also try to figure out where the concurrent creation of locks is coming from and avoid that instead.